interfaces/builtin/packagekit_control.go

```diff
@@ -77,24 +77,24 @@ dbus (send)
 # packagekitd) changes this, then these rules will need to change too.
 dbus (receive, send)
         bus=system
-        path=/[0-9][0-9]*_[0-9a-f][0-9a-f]*
+        path=/[0-9]*_[0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f]
```

Comment:
Thank you for tightening this up. It made me think that if we knew what the largest number is for before '_', we could drop the '*' and therefore avoid accidentally matching /0foo/bar_deadbeef. Eg, if it corresponds to a pid, could use:
```
path=/{[1-9],[1-9][0-9],[1-9][0-9][0-9],[1-9][0-9][0-9][0-9],[1-9][0-9][0-9][0-9][0-9],[1-9][0-9][0-9][0-9][0-9][0-9],[1-4][0-9][0-9][0-9][0-9][0-9][0-9]}_[0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f]
```

Reply:
It is an auto-incrementing sqlite table primary key, so it potentially has the whole range of 64-bit integers.  I think the `/0foo/bar_deadbeef` case should be safe since I haven't used `**` though, right?  It should still only match single component object paths.

Reply:
>  is an auto-incrementing sqlite table primary key, so it potentially has the whole range of 64-bit integers. I think the `/0foo/bar_deadbeef` case should be safe since I haven't used `**` though, right? It should still only match single component object paths.

Yes (I considered that in my previous comment but not this one. sigh).

`path=/[0-9]*_[0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f]` is as good as we can do and we don't need to anguish over it since the interface is superprivileged and grants installation of packages which could rewrite the policy, etc, etc, etc. Thank you for checking the implementation.